internal/crypto/ptls/default.go

```diff
@@ -57,16 +192,54 @@ func Default(rootCAs *x509.CertPool) *tls.Config {
 	}
 }
 
+// buildCipherSuites returns the intersection of its parameters, as a list of cipher suite IDs.
+// If allowedCipherSuites is empty, it will return all the configuredCipherSuites.
+func buildCipherSuites(
+	configuredCipherSuites []*tls.CipherSuite,
+	allowedCipherSuites []*tls.CipherSuite,
+) []uint16 {
+	allowedCipherSuiteIDs := make([]uint16, 0)
+
+	for _, allowedCipherSuite := range allowedCipherSuites {
```

Comment:
Would the set and set intersection helpers be useful here?

Reply:
yep, done with next push